paddle/majel/malloc.cc

```diff
@@ -0,0 +1,131 @@
+#include "malloc.h"
+#include <glog/logging.h>
+
+#ifndef PADDLE_ONLY_CPU
+#include <cuda_runtime.h>
+#endif
+
+#define CHECK_CUDA(cudaFunc)                                         \
+  do {                                                               \
+    cudaError_t cudaStat = cudaFunc;                                 \
+    CHECK_EQ(cudaSuccess, cudaStat) << "Cuda Error: "                \
+                                    << cudaGetErrorString(cudaStat); \
+  } while (0)
+
+namespace majel {
+namespace malloc {
+namespace detail {
+#ifndef PADDLE_ONLY_CPU
+const char* get_device_error_string() {
+  cudaError_t err = cudaGetLastError();
+  return cudaGetErrorString(err);
+}
+
+const char* get_device_error_string(size_t err) {
+  return cudaGetErrorString((cudaError_t)err);
+}
+
+void* malloc_device(size_t size) {
```

Comment:
How about rename `malloc_device` into `malloc_cuda`? We have a plan to support other device interfaces like OpenCL and FPGA.